src/bake/production.zig

```diff
@@ -183,23 +199,50 @@ pub fn buildCommand(ctx: bun.CLI.Command.Context) !void {
 
     var root_dir_buf: bun.PathBuffer = undefined;
     const root_dir_path = bun.path.joinAbsStringBuf(cwd, &root_dir_buf, &.{"dist"}, .auto);
-    const root_path_trailing = root_dir_path.ptr[0 .. root_dir_path.len + 1];
-    _ = root_path_trailing; // autofix
-    root_dir_buf[root_dir_path.len] = std.fs.path.sep;
-    // server_bundler.options.public_path = root_path_trailing;
-    // server_bundler.resolver.opts.public_path = root_path_trailing;
-
-    var entry_points = std.ArrayList(BakeEntryPoint).init(allocator);
-    // the ordering of these entrypoints is relied on when inspecting the output chunks.
-    try entry_points.append(BakeEntryPoint.init(framework.entry_server, .server));
-    try entry_points.append(BakeEntryPoint.initClientWrapped(framework.entry_client, .client));
-
-    for (options.routes) |route| {
-        try entry_points.append(BakeEntryPoint.init(route.entry_point, .server));
+    // const root_path_trailing = root_dir_path.ptr[0 .. root_dir_path.len + 1];
+    // root_dir_buf[root_dir_path.len] = std.fs.path.sep;
+    // _ = root_path_trailing;
+
+    var router_types = try std.ArrayListUnmanaged(FrameworkRouter.Type).initCapacity(allocator, options.framework.file_system_router_types.len);
+
+    var path_map: ProductionPathMap = .{
+        .allocator = allocator,
+        .entry_points = .{},
+        .route_files = .{},
+        .client_files = .{},
+        .out_index_map = &.{},
+        .out_module_keys = &.{},
+    };
+
+    for (options.framework.file_system_router_types, 0..) |fsr, i| {
+        const joined_root = bun.path.joinAbs(cwd, .auto, fsr.root);
```

Comment:
the buffer used by `joinAbs` is guaranteed to not be used by `readDirInfoIgnoreError`?